framework/data/ActiveDataProvider.php

```diff
@@ -88,7 +94,7 @@ public function init()
     {
         parent::init();
         if ($this->db !== null) {
-            $this->db = Instance::ensure($this->db, Connection::className());
+            $this->db = Instance::ensure($this->db, $this->connectionClass);
```

Comment:
How about just removing the check? That would both pass the test *and* won't cause BC break.

```suggestion
            $this->db = Instance::ensure($this->db);
```